### ISSUE TITLE
"Red box of shame" on images without alts

### ISSUE THREAD

**Issue by nolanlawson** [nolanlawson/pinafore#2250]
This is a neat idea I heard that at least one instance did: put a red border around images without alts so folks know not to boost them.

Some concerns:

1. Maybe this should be an optional feature? If you don't know what the red border is and you suddenly start seeing them, it could be confusing.
2. Is there another way to encourage good behavior with images/alts? E.g. a dialog when you tap the boost button that says, "This imagine doesn't have an alt – are you sure you want to boost it?"
3. What about replying to the image with auto-generated text from Tesseract.js? I see a lot of people doing this informally. Could work some % of the time at least.

**Comment by MarcoZehe** [comment]
I like the idea about asking if one really wants to boost it. Also, it would be cool if Pinafore would hint at adding an image description if someone wants to post media that doesn't have a description yet. If it doesn't already. Sorry, I only post media with descriptions, so I don't know if Pinafore already hints at missing descriptions.

**Comment by PratikP1** [comment]
I like all options here. My concern with the the red boarder option is the potential for mistakes for colorblind people. Using a different color or another visual indication  could be more helpful. A combination of methods could be used to avoid solely relying on color.

How good are auto-generated alt texts from Tesseract.js? does anyone know?

**Comment by nolanlawson** [comment]
@MarcoZehe Pinafore does have a pretty prominent "description" `textarea` overlaying the image when composing a toot. It doesn't look exactly like the Mastodon v4 UI (which has a little :information_source: icon and the text "no description added") but IMO it seems to be roughly equally prominent.

@PratikP1 That is a good point, @NickColley has been arguing that we should not use (only) color to communicate anything, so this would fall in that category.

**Comment by NickColley** [comment]
We should not shame those who don't write alt text because while many do out of ignorance, or worse known apathy, some disabled people cannot write alt text.

We should instead make it easier to write alt text in pinafore that looks like:

- Showing alt text button in images that have it so you can see what have them and what don't
- opt-out reminder to add alt text that stops you from posting until if you forgot (which disabled people who can't can opt out of in settings)

**Comment by nolanlawson** [comment]
> Showing alt text button in images that have it so you can see what have them and what don't

This reminds me of https://github.com/nolanlawson/pinafore/issues/2102 ; having a button that shows the alt text would also double as an indication that the image has alt text.

> opt-out reminder to add alt text that stops you from posting until if you forgot (which disabled people who can't can opt out of in settings)

I'm a bit nervous to make it opt-out, but you know what, let's do it. It's good for folks to be reminded to post alt text.

We can add an option in the settings like "Allow omitting descriptions when posting images" (note that I tend to prefer options to be `false` by default, hence "allow omitting").

**Comment by siakaramalegos** [comment]
Here's a screenshot of what it looks like on front-end.social when an image description is missing (not trying to shame that person here). I like it because it reminds me to add descriptions and also I can quickly see when they're missing so I can boost them less

![Screenshot_20221125-143318](https://user-images.githubusercontent.com/4777393/204047234-8b592009-405d-4bd8-af13-a24697749f4c.png)

**Comment by Magess** [comment]
I would like the option to turn off such a border, if that's what you implement.

**Comment by Heartade** [comment]
> We should instead make it easier to write alt text in pinafore that looks like:
> 
>     * Showing alt text button in images that have it so you can see what have them and what don't
> 
>     * opt-out reminder to add alt text that stops you from posting until if you forgot (which disabled people who can't can opt out of in settings)

I love this idea, especially the button approach as the red border looks a little too intrusive to me and people could confuse it for having a serious error.

To add a bit of my own suggestion:

 * Media with alt text has a 'show alt text' button, which shows the alt text when pressed.
 * WIth media missing alt text, the 'show alt text' button is replaced with a 'no alt text' icon. The 'show alt text' and 'no alt text' icons should be distinguishable for people with color blindness, so an indicator design like the retweet/like buttons would be nice.

Considering a post may have multiple media attached, having a button per each media could look too crowded. I would prefer if there is one button for all media in a post (like the show/hide toggle), and pressing it expands the first media with its alt text shown.

By the way, would we need to distinguish between posts having _'no alt text at all'_ and _'some media missing alt text'_?
